CMakeLists.txt

```diff
@@ -15,7 +15,7 @@
 # limitations under the License.
 #
 
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.12)
```

Comment:
Introduced some useful macros which removed the need for our own versions. I think it may have also introduced the `glob(CONFIGURE_DEPENDS)` which reruns glob if cmake detects file changes (glob but without the bad bits)

Reply:
![image](https://user-images.githubusercontent.com/48598630/128704169-1705a775-9598-4e5d-bce8-e9688c7bc3c8.png)